src/controllers/searchController.ts

```diff
@@ -49,8 +50,40 @@ async function searchStalls(req: Request, res: Response, next: NextFunction) {
       return acc;
     }, []);
     const stalls = await findStallsByIdsFunc(stallIdsArray);
-    const formattedStalls = mapStallToCardFunc(stalls);
-    res.status(200).json(formattedStalls);
+    req.stalls = stalls;
+    next();
+  } catch (err) {
+    next(err);
+  }
+}
+
+/**
+ * Formats stall information to display them on cards.
+ * @param stalls Sequelize instances of stalls to be formmatted.
+ */
+async function mapStallToCard(req: Request, res: Response, next: NextFunction) {
```

Comment:
This should be the last middleware in the chain, because it is formatting output specifically for frontend.

Reply:
No other followup middlewares will use this.